conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosNodeOperator.scala

```diff
@@ -338,24 +338,60 @@ private[tezos] class TezosNodeOperator(
       (fetchCurrentQuorum, fetchCurrentProposal).mapN(CurrentVotes.apply)
     }
 
-  /** Fetches proposals for given blocks */
+  /** Fetches proposals for given blocks
+    * The returned value, when available for active voting periods, is the ID of
+    * a proposal, though for the sake of naming consistency with tezos schema
+    * we refer to that as a protocol ID.
+    */
   def getProposals(blocks: List[BlockData]): Future[List[(BlockHash, Option[ProtocolId])]] = {
     import cats.instances.future._
     import cats.instances.list._
     import tech.cryptonomic.conseil.common.generic.chain.DataFetcher.fetch
-    fetch[BlockHash, Option[ProtocolId], Future, List, Throwable].run(blocks.filterNot(b => isGenesis(b)).map(_.hash))
+
+    val nonGenesisHashes = blocks.filterNot(isGenesis).map(_.hash)
+    fetch[BlockHash, Option[ProtocolId], Future, List, Throwable].run(nonGenesisHashes)
   }
 
-  /** Fetches detailed data for voting associated to the passed-in blocks */
-  def getVotingDetails(blocks: List[Block]): Future[List[BakerBlock]] = {
+  /** Fetches rolls for all bakers of individual blocks
+    *
+    * @param blockHashes defines the blocks we want the rolls for
+    * @return the lists of rolls, for each individual requested hash
+    */
+  def getBakerRollsForBlockHashes(blockHashes: List[BlockHash]): Future[List[(BlockHash, List[Voting.BakerRolls])]] = {
     import cats.instances.future._
     import cats.instances.list._
     import tech.cryptonomic.conseil.common.generic.chain.DataFetcher.fetch
 
-    val fetchBakers =
-      fetch[Block, List[Voting.BakerRolls], Future, List, Throwable]
+    fetch[BlockHash, List[Voting.BakerRolls], Future, List, Throwable].run(blockHashes)
+  }
+
+  /** Fetches rolls for all bakers of individual blocks
+    *
+    * @param blocks the blocks we want the rolls for
+    * @return the lists of rolls, for each individual requested block
+    */
+  def getBakerRollsForBlocks(blocks: List[Block]): Future[List[BakerBlock]] = {
```

Comment:
`BakerBlock` is ambiguous as well. Can we rename the type to something like `BakerRollsByBlock`?